sim/src/mechanics/walking.rs

```diff
@@ -932,3 +977,38 @@ impl IndexableKey for PedestrianID {
         self.0
     }
 }
+
+/// Returns a number in (0, 1] to multiply speed by to account for current crowdedness.
```

Comment:
Most of the PR is boilerplate to make the UI handle the new problem type. The interesting bit is in here